File: metro_tracker_app/lib/features/metro_tracking/domain/entities/station.dart
import 'package:equatable/equatable.dart';

class Station extends Equatable {
  final String id;
  final String name;
  final String code;
  final int orderIndex;
  final Coordinates? coordinates;
  final double? distance; // Optional, for nearby stations

  const Station({
    required this.id,
    required this.name,
    required this.code,
    required this.orderIndex,
    this.coordinates,
    this.distance,
  });

  @override
  List<Object?> get props =>
      [id, name, code, orderIndex, coordinates, distance];
}

class Coordinates extends Equatable {
  final double lat;
  final double lng;

  const Coordinates({
    required this.lat,
    required this.lng,
  });

  @override
  List<Object?> get props => [lat, lng];
}

File: backend/src/models/Station.js
const mongoose = require('mongoose');

const stationSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true
    },
    code: {
        type: String,
        required: true,
        unique: true,
        uppercase: true,
        trim: true
    },
    orderIndex: {
        type: Number,
        required: true
    },
    // Optional: for future map integration
    coordinates: {
        lat: Number,
        lng: Number
    }
}, {
    timestamps: true
});

// Index for faster queries

stationSchema.index({ orderIndex: 1 });

module.exports = mongoose.model('Station', stationSchema);
